framework/include/kernels/ConservativeAdvection.h

```diff
@@ -28,12 +28,25 @@ class ConservativeAdvectionTempl : public GenericKernel<is_ad>
 protected:
   virtual GenericReal<is_ad> computeQpResidual() override;
   virtual Real computeQpJacobian() override;
+  virtual Real computeQpOffDiagJacobian(unsigned int jvar) override;
   virtual void computeResidual() override;
   virtual void computeJacobian() override;
 
+  /// Scalar material property
+  const GenericMaterialProperty<Real, is_ad> & _scalar;
+
+  /// field variable
+  const VariableGradient * _field_variable;
```

Comment:
The doxygen and name `_field_variable` make me think this will be a `MooseVariable` instead of a `VariableGradient`. I would probably rename